pkg/platform/kube/functionres/lazy.go

```diff
@@ -2159,31 +2316,31 @@ func (lc *lazyClient) populateDeploymentContainer(ctx context.Context,
 	}
 }
 
-func (lc *lazyClient) populateSidecarContainer(ctx context.Context,
-	sidecarSpec *v1.Container,
+func (lc *lazyClient) populateContainer(ctx context.Context,
```

Comment:
I actually think we can overhaul this function and just put the spec in the actual container as is, because by populating the fields manually we can miss other fields set in the container spec (my fault, I know 🙈 ).
We would only need to call `EnrichSupplementaryContainerResources`, and deal with the rest of the enrichment earlier in the dashboard (as is already done).

Reply:
@TomerShor make sense, removed this method at all